cypress/e2e/app/simple-e2etest.cy.js

```diff
@@ -37,30 +37,32 @@ describe("End to end test using a simple UI path through the app", () => {
                 ]
             };
 
-            if ( cy.datasetMeetsTestCriteria("annotation", p_dataset, testCriteria) ) {
+            // 1. Go through index page, selecting participants tsv and json dictionary files
 
-                // 1. Go through index page, selecting participants tsv and json dictionary files
+            // A. Assert that categorization nav item and next button are disabled
+            cy.assertNextPageAccess("categorization", false);
 
-                // A. Assert that categorization nav item and next button are disabled
-                cy.assertNextPageAccess("categorization", false);
+            // B. Select data table file
+            cy.get("[data-cy='data-table-selector']")
+                .contains("Choose file")
+                .click()
+                .selectFile(dataFolder + p_dataset.data_table);
 
-                // B. Select data table file
-                cy.get("[data-cy='data-table-selector']")
-                    .contains("Choose file")
-                    .click()
-                    .selectFile(dataFolder + p_dataset.data_table);
+            // C. Assert that categorization nav item and next button are enabled
+            cy.assertNextPageAccess("categorization", true);
 
-                // C. Assert that categorization nav item and next button are enabled
-                cy.assertNextPageAccess("categorization", true);
+            // D. Select participants dictionary
+            cy.get("[data-cy='data-dictionary-selector']")
+                .contains("Choose file")
+                .click()
+                .selectFile(dataFolder + p_dataset.data_dictionary);
 
-                // D. Select participants dictionary
-                cy.get("[data-cy='data-dictionary-selector']")
-                    .contains("Choose file")
-                    .click()
-                    .selectFile(dataFolder + p_dataset.data_dictionary);
+            // E. Click the next page button to proceed to the categorization page
+            cy.nextPageByButton();
 
-                // E. Click the next page button to proceed to the categorization page
-                cy.nextPageByButton();
+            cy.commitToVuexStore("setCurrentPage", "categorization");
```

Comment:
mentioned this for the `annotation-pagetest` as well: if this line is needed, there should be a comment to explain why. Or, if we keep re-using this pattern, we might have to add a central README where we explain this. 

For example, here we're going to the next page via the helperFunction `nextPageByButton`, and still have to explicitly set the `setCurrentPage` mutation. In the annotation page test, we use `router.push`. I'm not sure if both paths are the same under the hood (and ideally I don't have to go and read the helper function to understand the test) so a comment is probably warranted.

Before you add it, just check that we do in fact have to set this line here, because if we can remove it, we'll save two lines :)

Reply:
I recall setting these (in both tests) to get the right button. Without setting the current page explicitly, it seems like there is a routing/store update issue somewhere that would give the wrong current and thus the wrong nextPage e.g., the e2e test was running in the categorization page but it would get the annotation page `next-page` button. I looked at the other e2e tests namely the categorization page and found this to be the solution.
@jarmoza Did you run into this issue as well? or rather do you know why we have to explicitly set the current page?

Reply:
Looking into this.

Reply:
@surchs and @rmanaem I think I could use a second eye on this and/or will take a look with a fresh set of eyes tomorrow.

I've debugged down to some unusual behavior on the part of Cypress. In the below screen shot is from the Chrome debugger. I've commented out anything after moving to the categorization page in test and put in print statements as the test proceeds. Most importantly, I've commented out the `commitToVuexStore` call (what we think should be redundant).

<img width="1437" alt="Screen Shot 2023-04-19 at 5 35 54 PM" src="https://user-images.githubusercontent.com/4140162/233205758-0a30d5fd-3e04-4e7e-8db3-56364ee4a5fe.png">

What we see is the function proceeding as normally and then once the end of the test is reached, the asynchronous calls are made – namely the call to `nextPageByButton` call.

I noticed that `setCurrentPage` is being called *twice* afterward. To check where that's happening, I wrapped the two places where `setCurrentPage` is called. Above you can see the next page button being identified as where `setCurrentPage` is being called from.

Unless I am missing something this seems like Cypress is firing the click event on the next page button twice.

Reply:
This is consistent with what I saw during refactoring.

Reply:
Taking another look at this morning.

Reply:
The discussed `next-page` component test will be placed in a separate PR. It's almost ready, but regardless will not change the outcome here. We have this double-click issue with Cypress occurring, and have decieded to keep the `commitToVueXStore` calls that prevent the side effect of the test-based, extra click event from affecting the outcome of the test.